src/stub/parser.rs

```diff
@@ -127,7 +110,15 @@ impl<'a> Parser<'a> {
                     let term_name = term.trim_matches(|c| c != '"').trim_matches('"').to_string();
                     JoinTerm::new_literal(term_name)
                 } else {
-                    JoinTerm::new_variable(term.trim().to_string())
+                    let ident = term.trim().to_string();
+                    match self.read_pairings.get(&ident) {
```

Comment:
Since all `None` does is panic, this should probably be an `expect`.

Reply:
```
                    let var_type = self.read_pairings.get(&ident).unwrap_or_else(|| {
                        panic!("The JoinTerm '{}' was not previously initialized.", &ident)
                    });
                    JoinTerm::new(ident, Some(*var_type))
```
I'm not sure this is better in terms of readability.